d2/src/main/java/com/linkedin/d2/discovery/event/PropertyEventBusImpl.java

```diff
@@ -188,8 +188,7 @@ public void publishInitialize(final String prop, final T value)
   {
     if (value == null)
     {
-      _log.warn("Received a null event during publishInitialize for String prop = " + prop +
-                    ". Still publishing the null event.");
+      return; // never publish null value
```

Comment:
Just thought about another case where there could be some subscribers waiting for this data. Instead of just return, we should call onAdd/onInitialize with the current value in _properties. 

There is chance that the current value in _properties is also null, in this case, we should add a new method "onReceiveNull(final String propertyName)" in AbstractLoadBalancerSubscriber:
```
@Override
  public void onReceiveNull(final String propertyName)
  {
    trace(_log, _name, ".onReceiveNull: ", propertyName);

    for (LoadBalancerState.LoadBalancerStateListenerCallback waiter : _waiters.get(propertyName).close())
    {
      waiter.done(_type, propertyName);
    }
  }
```

Note we don't call handlePut in this method (so all simple load balancer state won't change), just notify all waiters (callbacks), if there is any, to complete.

Reply:
Thanks. Update the code, will nofity all the wariters to complete the progress.